src/components/Main/SinglePostPage.jsx

```diff
@@ -0,0 +1,47 @@
+import { useLocation, useNavigate, useParams, } from "react-router-dom";
+import ProfileIcon from "../Header/ProfileIcon";
+import { useEffect, useState  } from "react";
+
+export default function SinglePostPage() {
+    const { id } = useParams()
+    const location = useLocation()
+    const navigate = useNavigate()
+ 
+    const [post, setPost] = useState(null)
+    
+    useEffect(() => {
+        if (location.state) {
+          setPost(location.state.post)
+        }
+      }, [location])
+
+      console.log(post)
+
+          if (post) {
+            return (
+              <div className="single-post-container">
+                <div className="post-header">
+                  <h3>{post.title}</h3>
+                  <p>{post.content}</p>
+                  <button  onClick={() => navigate(`/posts`)} className="btn">
+                    Go Back
+                  </button>
+                </div>
+              </div>
+            )
+          }
+
+
+    
+
+
+
+
+
+
+
+
+
+ 
+    
```

Comment:
Your work is fantastic, good job. The only real feedback I have for you is to try and bit a little tidier - do you need 13 lines of whitespace here?

Reply:
Thank you for your feedback, and you are nice to me.
I don't know what happened, but I got lost somewhere, even tho I understood the concepts.
I promise that I will redo this challenge, and I'll make it awesome 💪